kroxylicious-integration-tests/src/test/java/io/kroxylicious/proxy/SaslInitiationIT.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.proxy;
+
+import java.time.Duration;
+import java.util.Map;
+import java.util.Set;
+
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.common.serialization.Serdes;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import io.github.nettyplus.leakdetector.junit.NettyLeakDetectorExtension;
+
+import io.kroxylicious.proxy.config.NamedFilterDefinition;
+import io.kroxylicious.proxy.config.NamedFilterDefinitionBuilder;
+import io.kroxylicious.proxy.testplugins.ClientTlsAwareLawyer;
+import io.kroxylicious.proxy.testplugins.SaslPlainInitiation;
+import io.kroxylicious.testing.kafka.api.KafkaCluster;
+import io.kroxylicious.testing.kafka.common.SaslMechanism;
+import io.kroxylicious.testing.kafka.junit5ext.KafkaClusterExtension;
+import io.kroxylicious.testing.kafka.junit5ext.Topic;
+
+import static io.kroxylicious.test.tester.KroxyliciousConfigUtils.proxy;
+import static io.kroxylicious.test.tester.KroxyliciousTesters.kroxyliciousTester;
+import static org.apache.kafka.clients.consumer.ConsumerConfig.AUTO_OFFSET_RESET_CONFIG;
+import static org.apache.kafka.clients.consumer.ConsumerConfig.GROUP_ID_CONFIG;
+import static org.apache.kafka.clients.producer.ProducerConfig.CLIENT_ID_CONFIG;
+import static org.apache.kafka.clients.producer.ProducerConfig.DELIVERY_TIMEOUT_MS_CONFIG;
+import static org.assertj.core.api.Assertions.assertThat;
+
+@ExtendWith(KafkaClusterExtension.class)
+@ExtendWith(NettyLeakDetectorExtension.class)
+public class SaslInitiationIT {
+    @Test
+    void shouldInitiate(@SaslMechanism(principals = { @SaslMechanism.Principal(user = "alice", password = "alice-secret") }) KafkaCluster cluster,
+                        Topic topic)
+            throws Exception {
+        String testName = "shouldInitiate";
+
+        NamedFilterDefinition saslInitiation = new NamedFilterDefinitionBuilder(
+                SaslPlainInitiation.class.getName(),
+                SaslPlainInitiation.class.getName())
+                .build();
+        NamedFilterDefinition lawyer = new NamedFilterDefinitionBuilder(
```

Comment:
I'm unfamiliar with the word 'lawyer' in this context.  Could you enlighten me?>

Reply:
Could go with the Auditor naming? It's quite similar, recording some details about whats going through it. Maybe with a name that makes it clear it's a test auditor to not collide with any real implementation.

Reply:
I think having an `AuditLogging` filter (see #2481) and a `*Auditor` is too suggestive that they're related when they're not. Given these are all things in the integration tests we are free to rename them later, so I don't think we need to sweat this one.